nakama/nakama.lua

```diff
@@ -1,3409 +1,3375 @@
+-- @param timezone (string) The timezone set by the user.
+-- @param username (string) The username of the user's account.
+
+-- @param callback_or_retry_policy Optional callback function or optional retry policy used specifically for this call
+-- A coroutine is used and the result is returned if no callback function is provided.
+-- @param retry_policy_or_nil Optional retry policy used specifically for this call (if callback was provided in previous argument) or nil
+-- @return The result.
+function M.update_account(client, avatarUrl, displayName, langTag, location, timezone, username, callback_or_retry_policy, retry_policy_or_nil)
+	assert(client, "You must provide a client")
+	assert(not avatarUrl or type(avatarUrl) == "string", "Argument 'avatarUrl' must be 'nil' or of type 'string'")
+	assert(not displayName or type(displayName) == "string", "Argument 'displayName' must be 'nil' or of type 'string'")
+	assert(not langTag or type(langTag) == "string", "Argument 'langTag' must be 'nil' or of type 'string'")
+	assert(not location or type(location) == "string", "Argument 'location' must be 'nil' or of type 'string'")
+	assert(not timezone or type(timezone) == "string", "Argument 'timezone' must be 'nil' or of type 'string'")
+	assert(not username or type(username) == "string", "Argument 'username' must be 'nil' or of type 'string'")
+
+
+	local url_path = "/v2/account"
+
+	local query_params = {}
+
+	local post_data = nil
+	post_data = json.encode({
+	avatarUrl = avatarUrl,
+	displayName = displayName,
+	langTag = langTag,
+	location = location,
+	timezone = timezone,
+	username = username,
+	})
+
+	local callback
+	local retry_policy
+	if type(callback_or_retry_policy) == "function" then
+		callback = callback_or_retry_policy
+		retry_policy = retry_policy_or_nil
+	else
+		callback = nil
+		retry_policy = callback_or_retry_policy
+	end
+	return http(client, callback, url_path, query_params, "PUT", post_data, retry_policy, function(result)
+		return result
+	end)
+end
+
+--- authenticate_apple
+-- Authenticate a user with an Apple ID against the server.
+-- @param client Nakama client.
+-- @param token (string) The ID token received from Apple to validate.
+-- @param vars (object) Extra information that will be bundled in the session token.
+
+-- @param create_bool () Register the account if the user does not already exist.
+-- @param username_str () Set the username on the account at register. Must be unique.
+-- @param callback_or_retry_policy Optional callback function or optional retry policy used specifically for this call
+-- A coroutine is used and the result is returned if no callback function is provided.
+-- @param retry_policy_or_nil Optional retry policy used specifically for this call (if callback was provided in previous argument) or nil
+-- @return The result.
+function M.authenticate_apple(client, token, vars, create_bool, username_str, callback_or_retry_policy, retry_policy_or_nil)
```

Comment:
The ideas was that if you are not using a callback then you wouldn't have to use nil, but could instead set the retry policy instead of the callback. But I agree, from a readability perspective it is not so great. I'll change it!